VIR-118 Sijaisuus

nniemi opened the pull request:
Added feature to add Sijainen (Replacement)

How to test:
Add a PositionEmployee
Modify the employee, see the Toggle to activate replacement
Check that it is updated correctly

nniemi committed:
Add position name search

nniemi committed:
initila

nniemi committed:
Add implementation

nniemi committed:
Merge branch 'develop' into feature/VIR-118-sijaisuus

0x464e committed:
Refactor replacement employee logic

0x464e committed:
Misc fixes for errors and warnings

0x464e committed:
Fix me being stupid

nniemi committed:
Fix implementation

github-actions[bot] committed:
Fix code style issues with CSharpier

Peskarhu commented on frontend/public/locales/fi/translation.json:
```diff
@@ -43,7 +43,13 @@
+    "replacement_active": "Sijaisuus voimassa",
+    "replacement_details": "Sijaisuuden tiedot",
+    "replacement_start_date": "Sijaisuuden alkamispäivämäärä",
+    "replacement_end_date": "Sijaisuuden päättymispäivämäärä",
```
Not really related and can be easily fixed later, but just a note that I think every "...päivämäärä" should be just "...päivä". I.e. "start_date": "Virkaan asettamispäivämäärä", --> "Virkaan asettamispäivä" or "Sijaisuuden alkamispäivä"

Peskarhu commented on frontend/src/components/Modal/EmployeeModal.tsx:
```diff
@@ -1,41 +1,78 @@
     try {
       const data = { ...values, positionId: position.id };
+      console.log(data);
+      console.log(values);
```
Leftovers?

0x464e committed:
Fixes & cleanup

0x464e committed:
Sneak in some auth bugfixes
Accidentally double logged out, via event handler & also via if statement. Now just relying on the event handler

0x464e committed:
More fixes

0x464e committed:
Fix warning spam to console

0x464e commented:
It's basically ready to merge (yes, spaghetti, but what can you do right now...), just has some known issues and bugs still.

1. save button gets enabled if switch is toggled (nothing is actually changed)
![NVIDIA_Overlay_2024-12-01_05-13-44](https://github.com/user-attachments/assets/55cffe46-0a83-4186-8983-ab30e86aa019)

2. Switch remembers user change state if component is closed and re-opened (initial checked state should always be be based off if there is an replacement or not)
![NVIDIA_Overlay_2024-12-01_05-16-33](https://github.com/user-attachments/assets/f30031d0-fe3c-439d-8f8a-bddb791e2a16)

and probably more..
